contracts/vesting/src/vesting/account/mod.rs

```diff
@@ -20,17 +18,10 @@ const BALANCE_SUFFIX: &str = "ba";
 const PLEDGE_SUFFIX: &str = "bo";
 const GATEWAY_SUFFIX: &str = "ga";
 
-fn generate_storage_key(b: &[u8], storage: &dyn Storage) -> Result<String, ContractError> {
-    let mut rng = StdRng::seed_from_u64(b.iter().fold(0, |acc, x| acc + *x as u64));
-    // Be paranoid and check for collisions
-    loop {
-        let key = rng.next_u64().to_string();
-        let balance_key = format!("{}{}", key, BALANCE_SUFFIX);
-        let balance: Item<Uint128> = Item::new(&balance_key);
-        if balance.may_load(storage)?.is_none() {
-            return Ok(key);
-        }
-    }
+fn generate_storage_key(storage: &mut dyn Storage) -> Result<String, ContractError> {
+    let key = KEY.may_load(storage)?.unwrap_or(0) + 1;
+    KEY.save(storage, &key)?;
+    Ok(key.to_string())
```

Comment:
I get this is a temporary workaround, but why is the key saved now and previously was only checked for existence?

Reply:
It was checked, as it was random, so the check was supposed to check for collisions, since we're only ever incrementing this it should never collide